runtime/oti/bytecodewalk.h

```diff
@@ -102,6 +102,11 @@ base types: (in the 19bit class index field)
 #define BCV_TAG_BASE_ARRAY_OR_NULL			2		/* set bit means base type array */
 #define BCV_SPECIAL_INIT					4		/* set bit means special init object ("this" for <init>) */
 #define BCV_SPECIAL_NEW						8		/* set bit means special new object (PC offset in upper 28 bits) */
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+#define BCV_PRIMITIVE_VALUETYPE				16		/* set bit means primitive class */
+#endif /* #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
+
+#define BCV_GET_TYPE_FROM_CHAR(char) (IS_QTYPE(char) ? BCV_PRIMITIVE_VALUETYPE : BCV_OBJECT_OR_ARRAY)
```

Comment:
should be IS_QTYPE((char))

Reply:
addressed in 0c91935 along with your other comments. I also defined `BCV_GET_TYPE_FROM_CHAR(char)` as just `(BCV_OBJECT_OR_ARRAY)` in non-VT builds because `BCV_PRIMITIVE_VALUETYPE` isn't defined in that case.